src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -99,6 +100,19 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     context.Dispose();
                     throw CreateSslException(SR.net_allocate_ssl_context_failed);
                 }
+
+                if (certHandle != null && !certHandle.IsInvalid)
+                {
+                    using (X509Certificate2 cert = new X509Certificate2(certHandle.DangerousGetHandle()))
```

Comment:
Extremely unlikely, but is there anything that prevents certHandle from being collected and finalized after the call to DangerousGetHandle and before the X509Certificate2 ctor dups it?  There's no remaining usage of certHandle in this method, so the JIT tracking could mark it as unused, and if the caller isn't guaranteed to be keeping it alive for the duration of this method, it's theoretical possible such a race condition could happen, which would cause things to blow up.  The fix would be to put a call to GC.KeepAlive(certHandle) at the end of the if block.